src/components/input/input.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { customElement, LitElement, html, property } from 'lit-element';
+import classnames from 'classnames';
+import settings from 'carbon-components/es/globals/js/settings';
+import WarningFilled16 from '@carbon/icons/lib/warning--filled/16';
+import styles from './input.scss';
+
+const { prefix } = settings;
+
+/**
+ * Supported input types.
+ *
+ * For this component we only support textual types
+ */
+export enum INPUT_TYPE {
+  EMAIL = 'email',
+  PASSWORD = 'password',
+  TEL = 'tel',
+  TEXT = 'text',
+  URL = 'url',
+}
+
+/**
+ * Input element. Supports all the usual attributes for textual input types
+ */
+@customElement(`${prefix}-input`)
+export default class BXInput extends LitElement {
+  /**
+   * May be any of the standard HTML autocomplete options
+   */
+  @property()
+  autocomplete = '';
+
+  @property({ type: Boolean })
```

Comment:
It'd be nice if we can add comments to properties, even if it came from native `<input>` - Given `BXInput` is not technically inheriting `<input>` (due to lack of type extension), rather it's a wrapper.